### PR TITLE
fix: prevent path.join from creating malformed URL for client entry path

### PR DESCRIPTION
**Pull request by arian0zen**
When using path.join to construct the client entry path in a Vite application with a custom base path, client-side hydration fails. This prevents client-side JavaScript from loading, leading to non-functional click events and other interactivity.

This PR updates the joining method to fix this issue!

fiixes #287

### CODE REVIEW

**Comment by mosch** [comment]
Awesome, thanks for the quick fix. And thank you for your contribution!

**Review by mosch — APPROVED** [review]

File: packages/zudoku/src/vite/dev-server.ts [hunk 1]
```diff
@@ -44,12 +43,20 @@ export class DevServer {
...
+    const proxiedEntryClientPath =
+      (vite.config.base.endsWith("/")
+        ? vite.config.base
+        : vite.config.base + "/") + "__z/entry.client.tsx";
+
+    // Ensure the path uses forward slashes, even on Windows
+    const normalizedProxiedEntryClientPath = proxiedEntryClientPath.replace(
+      /\\/g,
+      "/",
     );
```
Comment (dan-lee) [inline packages/zudoku/src/vite/dev-server.ts]: Wouldn't `path.posix.join` suffice?

Comment (arian0zen) [reply packages/zudoku/src/vite/dev-server.ts]: ahh yeshh! That would be a better solution!
Updated the PR!

**Comment by mosch** [comment]
We're struggleing with the pipeline, sorry. We merged it manually https://github.com/zuplo/zudoku/pull/302

**Comment by arian0zen** [comment]
> We're struggleing with the pipeline, sorry. We merged it manually #302

is there any issue from my end ? shall I create a new branch in this repo itself rather than working on a new branch on the forked repo for the next feature that I might work on ?

**Pull request closed without merging**